plugin-ci: "Simulate App Store install" fails on Node 22 — brittle $(npm pack) capture polluted by prepare banner

### Summary

The **"Simulate App Store install (--ignore-scripts)"** step in `plugin-ci.yml` captures the packed tarball name with:

```bash
PLUGIN_TGZ=$(npm pack --ignore-scripts --pack-destination "$RUNNER_TEMP")
```

When the plugin under test has a `prepare` script (e.g. `"prepare": "tsc"` — the norm for TypeScript plugins), `npm pack` prints the lifecycle banner (`> pkg@ver prepare` / `> tsc`) to **stdout**, so `$PLUGIN_TGZ` ends up containing banner lines as well as the filename. The follow-up

```bash
npm install --ignore-scripts "$RUNNER_TEMP/$(basename "$PLUGIN_TGZ")"
```

then resolves a garbage path and fails with `ENOENT` / exit code 254.

### Why it only fails on Node 22

- **Node 24 (npm 11):** `--ignore-scripts` suppresses the pack-time `prepare`, so stdout is just the filename → step passes.
- **Node 22 (npm 10):** the pack-time `prepare` runs *despite* `--ignore-scripts`, and npm additionally reports the tarball as `(null) seems to be corrupted. Trying again.` → stdout is polluted and the step fails.

So every TypeScript plugin with `prepare: tsc` fails `plugin-ci` on the Node 22 matrix legs, while passing on Node 24. The published packages install fine via the real App Store (prebuilt tarball + `--ignore-scripts`, so `prepare` never runs) — this is a CI-only false negative.

### Observed failure (excerpt)

```
npm warn tarball signalk-noaa-sonar-charts-1.2.0.tgz (null) seems to be corrupted. Trying again.
npm error code ENOENT
npm error path > signalk-noaa-sonar-charts@1.2.0 prepare
npm error enoent ... signalk-noaa-sonar-charts-1.2.0.tgz'
##[error]Process completed with exit code 254.
```

Full run (Linux / Node 22 leg): https://github.com/joelkoz/signalk-noaa-sonar-charts/actions/runs/27524784081

### Suggested fix

Capture the filename robustly instead of trusting raw stdout — exactly as the **"Verify npm pack includes all required files"** step in this same workflow already does (around line 1107):

```bash
PLUGIN_TGZ=$(npm pack --ignore-scripts --json --pack-destination "$RUNNER_TEMP" | node -e 'let s="";process.stdin.on("data",d=>s+=d).on("end",()=>console.log(JSON.parse(s)[0].filename))')
```

(or any equivalent `--json` parse). This is npm-version-agnostic and unaffected by lifecycle-banner output. There are two occurrences of the brittle pattern to update — around **line 1294** (Simulate App Store install) and **line 1680**.

### Repro

A minimal plugin with `"prepare": "tsc"` (or any `prepare` that prints to stdout), run through `plugin-ci.yml` on the Node 22 matrix leg.

### Environment

- Workflow: `SignalK/signalk-server/.github/workflows/plugin-ci.yml@master`
- Introduced in #2628; brittle capture unchanged on master as of 2026-06-15.

--- a/.github/workflows/plugin-ci.yml
+++ b/.github/workflows/plugin-ci.yml
@@ -1291,7 +1291,11 @@ jobs:
           # This means native addons (better-sqlite3, serialport, etc.) are NOT compiled.
           # This test catches plugins that will break when installed via the App Store.
           echo "Simulating SignalK App Store install (npm install --ignore-scripts)..."
-          PLUGIN_TGZ=$(npm pack --ignore-scripts --pack-destination "$RUNNER_TEMP")
+          # Read the tarball name from --json: a plugin "prepare" script (e.g. tsc)
+          # prints a lifecycle banner to stdout that would pollute a bare
+          # $(npm pack) capture (npm < 11 runs prepare here despite --ignore-scripts).
+          PLUGIN_TGZ=$(npm pack --ignore-scripts --json --pack-destination "$RUNNER_TEMP" \
+            | node -e 'const s=require("fs").readFileSync(0,"utf8");process.stdout.write(JSON.parse(s.slice(s.search(/^\s*\[/m)))[0].filename)')
 
           mkdir -p "$APPSTORE_DIR"
           cd "$APPSTORE_DIR"
@@ -1675,9 +1679,11 @@ jobs:
 
       - name: Install plugin into SignalK
         run: |
-          # Pack the plugin and install it into the test server. Use basename
-          # defensively: some npm versions emit a path-prefixed tarball name.
-          PLUGIN_TGZ=$(npm pack --ignore-scripts --pack-destination /tmp)
+          # Pack the plugin and install it into the test server. Read the tarball
+          # name from --json so a plugin "prepare" lifecycle banner on stdout can't
+          # corrupt it (npm < 11 runs prepare here despite --ignore-scripts).
+          PLUGIN_TGZ=$(npm pack --ignore-scripts --json --pack-destination /tmp \
+            | node -e 'const s=require("fs").readFileSync(0,"utf8");process.stdout.write(JSON.parse(s.slice(s.search(/^\s*\[/m)))[0].filename)')
           cd /tmp/sk-test
           npm install "/tmp/$(basename "$PLUGIN_TGZ")"
 

